### PR TITLE
Update sqlflow.proto to support async submit SQLFlow job

### PR DESCRIPTION
**Pull request by Yancey0623**
part work of #1066

### CODE REVIEW

File: pkg/server/proto/sqlflow.proto [hunk 1]
```diff
@@ -17,6 +17,36 @@ service SQLFlow {
...
+    // Each of these SQL statments can be a standard SQL like:
+    // `SELECT ... FROM ...;`, `DESCRIBE ...`,
+    // or an extended SQLFlow SQL like:
+    // `SELECT ... TO TRAIN/PREDICT/EXPLAIN ...`.
+    //
+    // Submit returns a Job message which contains the SQLFlow Job ID.
+    rpc Submit (Request) returns (Job);
+    
+    // Fetch fetchs the SQLFlow job status and logs in a polling manner.
+    rpc Fetch (Job) returns(JobStatus);
```
Comment (weiguoz) [inline pkg/server/proto/sqlflow.proto:33]: Intuitively, considering the polling manner, I would suggest renaming `Fetch` to `Peek`.

File: pkg/server/proto/sqlflow.proto [hunk 2]
```diff
@@ -17,6 +17,36 @@ service SQLFlow {
...
+message JobStatus {
+    enum Code {
+        PENDING = 0;
+        RUNNING = 1;
+        SUCCEEDED = 2;
+        FAILED = 3;
+        UNKNOWN = 4;
+    }
+    // The message field indicates the job logs or error message.
+    string message = 1;
```
Comment (weiguoz) [inline pkg/server/proto/sqlflow.proto]: Is it lack of  `Code  status  = 1; // (or 2)` ?

**Review by typhoonzero — APPROVED** [review]
LGTM. Do we still use a "jobid" to query job status?

**Review by weiguoz — APPROVED** [review]
LGTM

**Pull request merged**